pkg/expect/expect.go

```diff
@@ -109,12 +110,18 @@ func (ep *ExpectProcess) ExpectFunc(f func(string) bool) (string, error) {
 				return line, nil
 			}
 		}
+		ep.mu.Unlock()
+
```

Comment:
This makes reading `ep.err` outside of lock. 

It's not obvious, but it should be ok? Even if comparing interfaces is not atomic, it should work as you are checking equality to nil. Maybe I'm wrong, but I hope you thought about this :P